napi/oxlint2/src-js/index.js

```diff
@@ -103,6 +105,24 @@ class Context {
       externalRuleId: this.#ruleId,
     });
   }
+
+  /**
+   * Get the absolute path of the file being linted.
+   * @deprecated Use `filename` instead.
```

Comment:
i don't think we should implement deprecated APIs, just for the sake of filling out the context object

Reply:
I figured these methods are part of ESLint's API, so we should include them, even though they're deprecated. An existing ESLint plugin might use these APIs and would break if they're absent.

Reply:
Lets chat in a bit? I don't quite agree, especially since we don't have full compat with eslint, i don't think we should fill out deprecated APIs, we shuold instead push plugin maintainers to use the new APIs

Reply:
yeah sorry to be a pain. I think let's close for now and return once it's confirmed.